manifests/base/applicationset-controller/argocd-applicationset-controller-role.yaml

```diff
@@ -62,4 +62,20 @@ rules:
     verbs:
       - get
       - list
-      - watch
\ No newline at end of file
+      - watch
+  # argocd-applicationset-controller leader election
+  - apiGroups:
+      - ''
+    resources:
+      - configmaps
```

Comment:
I'm not familiar with how leader election works in the appset controller. Can we limit this to one CM instead of all?

Reply:
I have dug a bit into the code of applicationset_controller.
It uses [sigs.k8s.io/controller-runtime@v0.11.0 manager.New (alias NewManager)](https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/manager#New) here in [cmd/argocd-applicationset-controller/commands/applicationset_controller.go](https://github.com/argoproj/argo-cd/blob/e2e0da7fcc4e6a5cf040117a3af9c15a7d0c267f/cmd/argocd-applicationset-controller/commands/applicationset_controller.go#L118C9-L118C9)

It sets `LeaderElectionID = "8ac56fa.applicationsets.argoproj.io"` so my assumption of a "random" hash was wrong. This means we can limit both configmap and lease to this specific name.

Will update the PR

Reply:
Also in `v0.11.0` the default for `sigs.k8s.io/controller-runtime` is `configmapsleases` which seems to be changed to `leases` in `v0.12.0`

According to the description of the latest docs it should be save to just change to `leases`.

In case I would change the value explicitly already we could remove the RBAC rule for ConfigMap from the PR.

> // So, what do you have to do when you are updating your controller-runtime dependency
	// from a lower version to v0.12 or newer?
	// - If your operator matches at least one of these conditions:
	//   - the LeaderElectionResourceLock in your operator has already been explicitly set to "leases"
	//   - the old controller-runtime version is between v0.7.0 and v0.11.x and the
	//     LeaderElectionResourceLock wasn't set or was set to "leases"/"configmapsleases"/"endpointsleases"
	//   feel free to update controller-runtime to v0.12 or newer.

https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager#New